S/Sundials/build_tarballs.jl

```diff
@@ -1,31 +1,37 @@
 using BinaryBuilder
 
 name = "Sundials"
-version = v"3.1.1"
+version = v"5.1.0"
 
 # Collection of sources required to build SundialsBuilder
 sources = [
     "https://github.com/LLNL/sundials/archive/v$(version).tar.gz" =>
-    "d03fca24d7adbfe52e05bd4e61340251e1cf56927540cc77a1ca817716091166",
+    "101be83221f9a0ab185ecce04d003ba38660cc71eb81b8a7cf96d1cc08b3d7f9",
     "./bundled",
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/sundials-*/
-patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
+if [[ "${target}" == *-mingw* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_windows.patch
+fi
 
 CMAKE_FLAGS=(-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}")
 CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release -DEXAMPLES_ENABLE_C=OFF)
 CMAKE_FLAGS+=(-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR="$prefix/include" -DKLU_LIBRARY_DIR="$libdir")
-CMAKE_FLAGS+=(-DBLAS_ENABLE=OFF -DLAPACK_ENABLE=OFF)
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
 
-#if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
-#    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas64_.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
-#else
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
-#fi
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
+export CFLAGS="-lgfortran"
+if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
+    export CFLAGS="${CFLAGS} -lquadmath"
+fi
```

Comment:
```sh
if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
    export CFLAGS="${CFLAGS} -lquadmath"
elif [[ "${target}" == powerpc64le-* ]]; then
    export CFLAGS="${CFLAGS} -Wl,-rpath-link,/opt/${target}/${target}/lib64"
fi
```
This partially solves the problem with PowerPC, but after this the lapack test fails with
```
/opt/bin/cc -lgfortran -Wl,-rpath-link,/opt/powerpc64le-linux-gnu/powerpc64le-linux-gnu/lib64 -O3 -DNDEBUG  -rdynamic CMakeFiles/ltest.dir/ltest.o  -o ltest -Wl,-rpath,/workspace/destdir/lib /workspace/destdir/lib/libopenblas64_.so 
/workspace/destdir/lib/libopenblas64_.so: undefined reference to `GOMP_parallel@GOMP_4.0'
collect2: error: ld returned 1 exit status
```
Indeed
```sh
sandbox:${WORKSPACE}/srcdir/sundials-5.1.0/build/LapackTest # nm /opt/${target}/${target}/lib64/libgomp.so | grep GOMP_parallel00000000000097b0 T GOMP_parallel_end
00000000000078e0 T GOMP_parallel_loop_dynamic_start
0000000000007900 T GOMP_parallel_loop_guided_start
0000000000007920 T GOMP_parallel_loop_runtime_start
00000000000078c0 T GOMP_parallel_loop_static_start
0000000000009bf0 T GOMP_parallel_sections_start
0000000000009730 T GOMP_parallel_start
```
there is no `GOMP_parallel` symbol.  This is a big issue.

Reply:
Ok, after an hour of trial and error I'm pretty confident that this should work for all flavours of PowerPC:
```sh
export CFLAGS="-lgfortran"
if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
    export CFLAGS="${CFLAGS} -lquadmath"
elif [[ "${target}" == powerpc64le-* ]]; then
    export CFLAGS="${CFLAGS} -lgomp -ldl -lm -lpthread -Wl,-rpath-link,/opt/${target}/${target}/lib64"
fi
```
but it needs also to change minimum GCC version, see next message